### ISSUE TITLE
Speed up `Graph#getEdgeIteratorState`

### ISSUE THREAD

**Issue by easbar** [graphhopper/graphhopper#1847]
Obtaining an `EdgeIteratorState` given an integer edge id is unnecessarily expensive, because `Graph#getEdgeIteratorState` creates an object instead of simply updating the edge id being in use as we do with the other `EdgeExplorer`s. We need something like `RandomAccessEdgeExplorer#setEdgeId` in analogy to `EdgeExplorer.setBaseNode`.

 Probably we should do this after #1780.

**Comment by michaz** [comment]
Can I ask something outrageous? Do you thing we could -- in the long run -- a) rename these things to *Cursor, and b) refactor them so that we can make them Iterable, like in the primitive-collections library we are using?

**Comment by easbar** [comment]
Doesn't sound so outrageous :) I don't really care about Iterator vs. Cursor what do you think makes the difference ? What do you mean making them iterable ? You mean having something like prev() in addition to next() ?

**Comment by michaz** [comment]
Yes. java.lang.iterable. The things in carrotsearch-collections: You can foreach them, even though they are cursors.

**Comment by michaz** [comment]
(I can‘t imagine how it could be slower than manually writing a hasNext()-next()-loop. In fact I think it‘s literally compiled to that.)
